jest/setup.js

```diff
@@ -13,3 +13,14 @@ jest.mock('../src/libs/Notification/PushNotification', () => ({
 }));
 
 jest.mock('react-native-blob-util', () => ({}));
+
+// Set up manual mocks for any Logging methods that are supposed hit the 'server',
+// this is needed because before, the Logging queue would get flushed while tests were running,
+// causing unexpected calls to HttpUtils.xhr() which would cause mock mismatches and flaky tests.
+/* eslint-disable no-console */
+jest.mock('../src/libs/Log', () => ({
```

Comment:
Sorry i didn't get why we're adding this here in this PR

Reply:
We need this for some of the jest unit tests to pass. Basically, we have a Log queue that gets updated every time we wanna log something on the server. This gets flushed once it reaches a certain length, causing a bunch of additional API calls to be made (and additional calls to `HttpsUtils.xhr`, which we mock). With the new API utility functions, we now update this Logging queue in our request middleware (this didn't happen before).

The logging queue can get flushed unexpectedly when some tests run, causing certain mocks to `HttpsUtils.xhr` to be mismatched (causing flaky tests). This block that I added basically mocks any calls to `Log` so that we won't invoke any additional API requests.

You can read more about this in [this Slack thread](https://expensify.slack.com/archives/C03TQ48KC/p1662633726757179).

TLDR it's needed to get the jest unit tests to pass, and without it they're flaky and confusing. We added this block so no one will run into this issue in the future. I spent almost a week trying to figure out what was going on

Reply:
Thanks for explaining. Makes sense, By this we make sure we flush the log queue before making the API call rather then it gets flushed in between and cause tests to be flaky, right? 🙇